pro/src/pages/AdageIframe/app/components/OffersInstantSearch/OffersSearch/Autocomplete/Autocomplete.tsx

```diff
@@ -261,21 +262,36 @@ export const Autocomplete = ({
             (elm) => elm.objectID === item.objectID
           )
 
-          // if the id is less than 3, the category is displayed and must be pre-selected in the filters.
-          if (
-            itemId >= 0 &&
-            itemId < 3 &&
-            item['offer.subcategoryId'] &&
-            item['offer.subcategoryId'].length > 0
-          ) {
-            const result = getCategoriesFromSubcategory(
-              item['offer.subcategoryId'][0]
-            )
-
-            result &&
-              (await formik.setFieldValue('categories', [result.subCategories]))
+          if (isFormatEnable) {
+            if (
+              itemId >= 0 &&
+              itemId < 3 &&
+              item.formats &&
+              item.formats.length > 0
+            ) {
+              await formik.setFieldValue('formats', [item.formats[0]])
+            } else {
+              await formik.setFieldValue('formats', [])
+            }
```

Comment:
(proposition) À voir si ça permets pas de simplifier encore plus le code avec le if d'après
```suggestion
            if (
              itemId >= 0 &&
              itemId < 3
            ) {
              await formik.setFieldValue('formats',  item.formats ? [item.formats[0]] : [])
            }
```

Reply:
on perd une fonctionnalité qui est que lorsqu'on clique sur la suggestion 4 ou 5 qui n'a pas de format/catégorie, on garde l'ancienne valeur alors que de base on veut la remettre à 0 d'où la ligne avec le else